### PR TITLE
Issue #37

### PR DESCRIPTION
**Pull request by chhavikirtani2000**
Implemented start_with along with tests.
@rubydesign Please review

### CODE REVIEW

**Comment by rubydesign** [comment]
Ok, the tests are getting there. If you could just split the method into several small ones, one assert per method. 
And basically never leave commented out code in the code base. Just remove and get it back with git's help if needed.

Now the tests pass, and do what we want, so far so good. 
But you have not removed the extra instance variables. As a general rule, you never need sinatce variables when they are only used in one method. Then they are local variables.

But you don't even need local (string) variables, only a counter.
And you don't need to_string , the word word instances support the [] operator just the same

**Comment by chhavikirtani2000** [comment]
Okay, thanks.
I did not understand how we can directly use word instance as if I try using [] operator on self, it gives an error?

File: lib/parfait/word.rb [hunk 1]
```diff
@@ -27,6 +28,8 @@ def self.get_length_index
     def initialize( len )
       super()
       @char_length = 0
+      #@str=str1
+      #@prefix=prefix1
```
Comment (rubydesign) [inline lib/parfait/word.rb]: these should go

File: lib/parfait/word.rb [hunk 2]
```diff
@@ -189,7 +192,20 @@ def to_s
     def padded_length
       Object.padded( 4 * get_type().instance_length + @char_length  )
     end
-
+    def start_with(other)
+      return false if other.length > self.length
+        s = other.length
+        temp = Word.new(other.length)
```
Comment (rubydesign) [inline lib/parfait/word.rb]: You don't need the temp variable
this can be done by just comparing other.get(i) with self.get(i) for all of others length

Comment (chhavikirtani2000) [reply lib/parfait/word.rb]: Okay, I updated. Please let me know any more improvements.

**Comment by rubydesign** [comment]
congrats @chhavikirtani2000 , you just finished your first pull request.

I just want to say that a lot of that was just from the first time. I'm sure i'll get easier.
And also, if you look at the last version compared to the first, i hope you can see that it is better code and worth the effort.

**Comment by chhavikirtani2000** [comment]
Thanks a lot for guiding me @rubydesign 
I learned a lot and would try to contribute more.

**Comment by rubydesign** [comment]
Sure, go for it. Didn't you say something about comments? (you can also just tell me where something is not explained enough)
But anything you are drawn to goes. Personally i would appreciate 35 most.

**Comment by chhavikirtani2000** [comment]
Sure! I will have a look into the codebase and do the needful.
Thank you.

**Pull request merged**